Prefer jpeg-turbo over jpeg

kleisauke opened the pull request:
It's 4x faster, better tested, and binary compatible. See the Homebrew discussion:
https://github.com/Homebrew/homebrew-core/pull/25682
https://github.com/Homebrew/homebrew-core/pull/25711

If you want, I can also change the `package.sh` build script (it should pack `jpeg-turbo` instead of `jpeg`).

kleisauke committed:
Prefer jpeg-turbo over jpeg

It's 4x faster, better tested, and binary compatible.

kleisauke committed:
Link explicitly with jpeg-turbo

Since the formula of jpeg-turbo is "keg-only".

lovell commented:
Hi Kleis, I vaguely recall looking into this in the past - there were a few linking problems in the dependency graph (libtiff also depends on libjpeg etc.) but I don't remember the details.

Please can you go ahead and make the changes to the package script too then I'll merge and we test the resultant file (it gets dumped into a private S3 bucket so I'll manually upload it here afterwards).

kleisauke committed:
Switch libtiff and gdk-pixbuf to jpeg-turbo

kleisauke committed:
Update package script

Use otool (recursively) to find shared libraries needed by libvips-cpp.42.dylib. This will only package the relevant shared libraries (unnecessary symlinks will not be included) and avoid manual removal of unused libraries.

kleisauke commented:
Indeed, I had forgotten that other dependencies also depend on libjpeg. It seems that only libtiff and gdk-pixbuf depend on libjpeg, so I also switched them to libjpeg-turbo with commit https://github.com/lovell/package-libvips-darwin/pull/3/commits/98cbd053e02d4423477920b1650935b43414689b.

I've enabled Travis on the master branch of my fork, and I see this on my [last build](https://travis-ci.org/kleisauke/package-libvips-darwin/builds/510260949#L4100):
```
libvips.42.dylib depends on libtiff.5.dylib
libtiff.5.dylib depends on libjpeg.8.dylib
...
libvips.42.dylib depends on libjpeg.8.dylib
...
libvips.42.dylib depends on librsvg-2.2.dylib
librsvg-2.2.dylib depends on libgdk_pixbuf-2.0.0.dylib
libgdk_pixbuf-2.0.0.dylib depends on libjpeg.8.dylib
```

So all dependencies that depend on libjpeg have been switched to libjpeg-turbo.

I also updated the package script (as requested) with commit https://github.com/lovell/package-libvips-darwin/pull/3/commits/31430e0328eb110278394a38d7b454f0092ca7ea. I used `otool` to find the shared libraries that were needed by `libvips-cpp.42.dylib`. This will package only the relevant shared libraries (unnecessary symlinks will not be included) and avoid manually deleting unused libraries. 

If you want to do the same for the Linux build you can use [this](https://github.com/kleisauke/libvips-packaging/blob/c1b2c142ffcc15367a571c0cb57a5f05cc08c87b/build/lin.sh#L301-L308). The only thing to take into account is that you should not link with symlinks in `binding.gyp`. I've checked the `*.dylib` libraries in `binding.gyp` and it seems to be fine.

Attachment: the tarball of the master branch ([`build #4`](https://travis-ci.org/kleisauke/package-libvips-darwin/builds/510260949)). I tested it on the libvips test suite and did not encounter any problems.
[libvips-8.7.4-darwin-x64.tar.gz](https://github.com/lovell/package-libvips-darwin/files/2999312/libvips-8.7.4-darwin-x64.tar.gz) (SHA1: 2c0ade46239daef4f901a698d480efabb020dc3d)

lovell commented:
Thanks for the updates Kleis, I'm on the road at the moment but should have time later this week to test the tarball.

Do you think we should move this repo into the libvips org? It started out as being sharp-specific but if you're able to use the output for your .net package then that feels like the right thing to do.

kleisauke commented:
> Do you think we should move this repo into the libvips org? It started out as being sharp-specific but if you're able to use the output for your .net package then that feels like the right thing to do.

Good idea! Maybe we should rename the repo to `homebrew-libvips` so that the following command works: `brew tap libvips/libvips`?

Do you think we should provide pre-compiled macOS binaries as part of the libvips release process? It can be useful for people who don't have Homebrew installed. FFmpeg also follows a similar approach: https://ffmpeg.zeranoe.com/builds/ (not sure if these binaries are "official" releases).

https://github.com/libvips/build-win64-mxe/issues/5 might be relevant here.

kleisauke committed:
gdk-pixbuf: remove introspection patch

See: Homebrew/homebrew-core#38790

kleisauke commented:
Quick update: the build was broken, so I fixed it with commit: https://github.com/lovell/package-libvips-darwin/pull/3/commits/00515e010253c3b017998bc443d0642d16ff094b.

Attachment: new tarball of the master branch ([`build #6`](https://travis-ci.org/kleisauke/package-libvips-darwin/builds/518177524#L4322)). Tested on the libvips test suite without any problems.
[libvips-8.7.4-darwin-x64.tar.gz](https://github.com/lovell/package-libvips-darwin/files/3063813/libvips-8.7.4-darwin-x64.tar.gz) (SHA1: 58cf032ce8c811802f5a75105fd2376c437c00fa)

lovell commented:
Thanks for the updates, will test them when I get a moment, probably next week. Once this is merged I'll move the repo and rename references so it can become an libvips org maintained "tap".

lovell commented:
Clean CI build of sharp on OS X using this tarball at https://travis-ci.org/lovell/sharp/builds/529945494 :+1:

kleisauke commented:
Thanks!

FYI: When testing the OS X binary files it sometimes causes failures with an obscure message:
```
Unable to revert mtime: /usr/local/share/fonts
```
See for example this Travis build: https://travis-ci.org/kleisauke/net-vips/jobs/518317733#L576. This is [already fixed](https://gitlab.freedesktop.org/fontconfig/fontconfig/merge_requests/7) within Fontconfig but not released yet.
